ui/src/app/services/server-config.service.ts

```diff
@@ -0,0 +1,21 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+
+import {ErrorHandlingService} from 'app/services/error-handling.service';
+
+import {ConfigResponse, ConfigService} from 'generated';
+
+@Injectable()
+export class ServerConfigService {
```

Comment:
@johnstill let me know if you have ideas on memoizing this `getConfig()` XHR with less boilerplate.

Reply:
I have some questions and thoughts about this that might be handled more easily via a slack call - hit me up when you're available.

Reply:
Good catch; the hot/cold subscription semantics are pretty dizzying. Per offline discussion and sandbox testing of behavior, `ReplaySubject` appears to be the right solution to this problem.

Reply:
Sorry, actually updated now. Previous push failed due to lint.